### ISSUE TITLE
Blacklisting of kbd engines removes some keyboards

### ISSUE THREAD

**Issue by njapke** [ibus/ibus-m17n#6]
In commit https://github.com/ibus/ibus-m17n/commit/289c8b738db5d424b796f0a9f79688d73b22280a some engines were blacklisted. The added comments already say something about better alternatives, but I don't quite understand why `m17n:*:kbd` was blocked, as it removes some keyboard layouts like the Cyrillic keyboard for Kazakh (`m17n:kk:kbd`) for which I haven't found an alternative. That very same keyboard even received a new symbol in commit https://github.com/ibus/ibus-m17n/commit/2463a70ba10bfab32f07b3a8394246244b548141 even though it should be inaccessible without commenting out the blacklisting.
Relevant code section (`src/default.xml`, lines 16-20):
```xml
<!-- kbd engines are duplicate of xkb engines -->
<engine>
	<name>m17n:*:kbd</name>
	<rank>-1</rank>
</engine>
```
The change seems to be quite old already, but since the 1.4.0 release is quite recent, it was only pushed into the repositories of my Linux distro (Arch Linux) quite recently, which is when the keyboard suddenly vanished for me. For the time being, I have commented out the cited code section to get it working again, but since this will possibly break with every update, I would like to ask, whether anything can be done here, or whether there are any alternative methods to get a Cyrillic keyboard layout for Kazakh in Ibus. I might also be missing something here, since the first referenced commit only blacklists these engines, because they are supposed to have clear alternatives.

**Comment by pnemade** [comment]
The comment says its duplicate of /usr/share/X11/xkb/symbols/kz keymap. I can select that xkb keyboard using ibus and input the same that is possible using /usr/share/m17n/kk-kbd.mim

**Comment by njapke** [comment]
So I was missing something. Thank you for the quick reply, then my system isn't configured properly and the proper keyboard isn't showing up. This issue can be closed then, as the problem is probably specific to my setup.

**Comment by njapke** [comment]
On closer research, I have now discovered, that the kz layout from XKB wasn't registered in Ibus, so it was never listed as an engine. Apparently, all XKB layouts used by Ibus are listen in `/usr/share/ibus/component/simple.xml`, and the Kazakh layout wasn't there for me. I have added it manually there now, and it works quite well, so I think I should take this issue to the normal Ibus repository, so that simple.xml can be changed, unless there is something else I'm missing, since the keyboard showed up for you, but I had to manually add it. I believe there are some other XKB layouts that aren't listed in simple.xml, so this might also be a problem for these.
